std/functional.d

```diff
@@ -1400,6 +1451,37 @@ unittest
     assert(fact(10) == 3628800);
 }
 
+// Issue 20099
+@system unittest // not @safe due to memoize
+{
+    int i = 3;
+    alias a = memoize!((n) => i + n);
+    alias b = memoize!((n) => i + n, 3);
+
+    assert(a(3) == 6);
+    assert(b(3) == 6);
+}
+
+@system unittest // not @safe due to memoize
+{
+    static Object objNum(int a) { return new Object(); }
```

Comment:
If you annotate `objNum` with `@safe` the test can be `@safe`.

Reply:
Nope, since `memoize`'s second overload isn't `@safe`.